components/command/utils.go

```diff
@@ -14,3 +16,12 @@ func WaitForCloudInit(runner *Runner) (*remote.Command, error) {
 			Sudo:   true,
 		})
 }
+
+func WaitUntilSuccess(runner *Runner) (*remote.Command, error) {
+	return runner.Command(
+		"wait-until-success",
+		&Args{
+			// echo works in shell and powershell
+			Create: pulumi.String("echo \"OK\""),
+		})
+}
```

Comment:
Maybe `WaitForEcho` ?

Reply:
I think `echo` is irrelevant here, we just wait for connection to success and we cannot do that without a command, so use a dummy command but it could any command really.
I will rename to `WaitForSuccessfulConnection`